ext/nnstreamer/tensor_decoder/tensordecutil.c

```diff
@@ -116,3 +117,25 @@ _free_labels (imglabel_t * data)
   data->total_labels = 0;
   data->max_word_length = 0;
 }
+
+/**
+ * @brief Copy framerate caps from tensor config
+ *
+ * @param[out]  caps to configure
+ * @param[in]   config to copy from
+ */
+void
+capsConfigFrameRate (GstCaps * caps, const GstTensorsConfig * config)
+{
+  gint fn, fd;
+
+  /** @todo Verify if this rate is ok */
+  fn = config->rate_n;
+  fd = config->rate_d;
+  if (!(fn >= 0 && fd > 0)) {
+    fn = 0;
+    fd = 1;
+  }
+  gst_caps_set_simple (caps, "framerate", GST_TYPE_FRACTION, fn, fd, NULL);
```

Comment:
If there is no "valid" framerate in config (i.e., the input caps has no framerate definition), why don't we keep it unconfigured?

Reply:
I actually wanted to hear any thoughts about this. I thought it might be safer to keep framerate caps without any particular reason. I will drop framerate caps if you think it's better.

Reply:
If there was no framerate caps in input, decoder should not make framerate caps in output.
If there was framerate caps in input, decoder should make the same framerate caps in output.

If there was framerate caps in output (reverse is possible sometiems), decoder should make the same framerate caps in input if it is trying to caps in reverse order.

Reply:
Thanks for explantion! I fixed it as you said. I probably need to get more experience in gstreamer :)